### PR TITLE
save wallet

### PR DESCRIPTION
**Pull request by melanke**
Saves the open wallet into the same file or a new file

### CODE REVIEW

File: src/NeoSharp.Core/Wallet/NEP6/NEP6WalletManager.cs [hunk 1]
```diff
@@ -272,10 +274,22 @@ private byte[] GetPrivateKeyFromWIF(string wif)
...
         }
 
-
+        /// <summary>
+        /// save the open wallet into the same file
+        /// </summary>
         public void SaveWallet()
         {
-            //throw new NotImplementedException();
+            SaveWallet(_openWalletFilename);
```
Comment (shargon) [inline src/NeoSharp.Core/Wallet/NEP6/NEP6WalletManager.cs:288]: check null ?

File: src/NeoSharp.Core/Wallet/NEP6/NEP6WalletManager.cs [hunk 2]
```diff
@@ -272,10 +274,22 @@ private byte[] GetPrivateKeyFromWIF(string wif)
...
+        }
+
+        /// <summary>
+        /// save the open wallet into a specific filename
+        /// </summary>
+        /// <param name="filename">the filename</param>
+        public void SaveWallet(String filename)
+        {
+            string json = _jsonConverter.SerializeObject(Wallet);
```
Comment (shargon) [inline src/NeoSharp.Core/Wallet/NEP6/NEP6WalletManager.cs]: check null?

File: src/NeoSharp.Application/Client/Prompt[Wallet].cs [hunk 3]
```diff
@@ -73,6 +73,20 @@ private void ImportNep2(string nep2key)
                 _consoleWriter.WriteLine(exception.Message, ConsoleOutputStyle.Error);
             }
         }
+        
+        [PromptCommand("wallet save", Category = "Wallet", Help = "Saves the open wallet into a new file")]
+        private void WalletSaveCommand(string fileName)
+        {
+            try
+            {
+                _walletManager.SaveWallet(fileName);
```
Comment (shargon) [inline src/NeoSharp.Application/Client/Prompt[Wallet].cs]: this try / catch is captured by Prompt, you could remove it

File: src/NeoSharp.Core/Wallet/NEP6/NEP6WalletManager.cs (hunk 2 above)
Comment (shargon) [reply src/NeoSharp.Core/Wallet/NEP6/NEP6WalletManager.cs]: filename could be null if is not open first

File: src/NeoSharp.Core/Wallet/NEP6/NEP6WalletManager.cs (hunk 1 above)
Comment (melanke) [reply src/NeoSharp.Core/Wallet/NEP6/NEP6WalletManager.cs:288]: Ok, I have checked on the other SaveWallet, so I think we are safe now

**Review by shargon — APPROVED** [review]

**Pull request merged**